src/index.rs

```diff
@@ -1395,6 +1395,18 @@ impl Index {
             } => self
                 .find_effective_type_info(referenced_type)
                 .unwrap_or(initial_type),
+            DataTypeInformation::String {
```

Comment:
this is a bit surprising for me.
is this always right? - also if you think about the contract of the method?
why is the intrinsic of STRING[10] == STRING[80]?

Reply:
I would feel better if this `special case` gets handled somewhere in the generics.rs